### PR TITLE
Location conversion tests for relay and parachains

### LINKED ISSUE #488: Integrate foreign locations to local accounts converter when upgrading polkadot-sdk to a newer version
Recently I introduced new foreign locations converter together with the tests to polkadot-sdk.
Those tests are also relevant to polkadot-fellows runtimes, thus creating this issue to track the implementation of the tests.
Relevant PR is [here](https://github.com/paritytech/polkadot-sdk/pull/5765)

### PR DESCRIPTION
**Pull request by x3c41a**
Added location convertion tests. 
Polkadot sdk [PR](https://github.com/paritytech/polkadot-sdk/pull/5765)
Fixes #488

### CODE REVIEW

File: system-parachains/asset-hubs/asset-hub-kusama/tests/tests.rs [hunk 1]
```diff
@@ -706,3 +708,118 @@ fn treasury_pallet_account_not_none() {
...
+						id: polkadot_core_primitives::AccountId::from(ALICE).into(),
+					},
+				],
+			),
+			expected_account_id_str: "5DGRXLYwWGce7wvm14vX1Ms4Vf118FSWQbJkyQigY2pfm6bg",
+		},
+		// DescribeAccountKey20Terminal
+		TestCase {
+			description: "DescribeAccountKey20Terminal Parent",
+			location: Location::new(1, [AccountKey20 { network: None, key: [0u8; 20] }]),
```
Comment (acatangiu) [inline system-parachains/asset-hubs/asset-hub-kusama/tests/tests.rs:773]: Same for these `AccountKey20`s:

```suggestion
			location: Location::new(1, [bob_20.clone()]),
```
with 
```rust
let bob_20 = AccountKey20 { network: None, key: [123u8; 20] }
```

(notice I also suggested using non-zero keys)

**Review by acatangiu — CHANGES_REQUESTED** [review]
The conversions are the same for all system parachains, so this test could be a macro to dedup the code, but I don't know where we would put that macro or if it's worth the effort. I'm fine with the duplicated code too.

Please see the other comments.

File: relay/kusama/tests/location_conversion.rs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+			expected_account_id_str: "5EBRMTBkDisEXsaN283SRbzx9Xf2PXwUxxFCJohSGo4jYe6B",
+		},
+		TestCase {
+			description: "DescribeBodyTerminal Sibling",
+			location: Location::new(
+				1,
+				[Parachain(1111), Plurality { id: BodyId::Unit, part: BodyPart::Voice }],
+			),
+			expected_account_id_str: "5DBoExvojy8tYnHgLL97phNH975CyT45PWTZEeGoBZfAyRMH",
+		},
```
[the remark this answers is not in the document]
Comment (x3c41a) [reply relay/kusama/tests/location_conversion.rs]: Done

File: relay/polkadot/tests/location_conversion.rs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+			expected_account_id_str: "5EBRMTBkDisEXsaN283SRbzx9Xf2PXwUxxFCJohSGo4jYe6B",
+		},
+		TestCase {
+			description: "DescribeBodyTerminal Sibling",
+			location: Location::new(
+				1,
+				[Parachain(1111), Plurality { id: BodyId::Unit, part: BodyPart::Voice }],
+			),
+			expected_account_id_str: "5DBoExvojy8tYnHgLL97phNH975CyT45PWTZEeGoBZfAyRMH",
+		},
```
[the remark this answers is not in the document]
Comment (x3c41a) [reply relay/polkadot/tests/location_conversion.rs]: ✅

File: system-parachains/asset-hubs/asset-hub-kusama/tests/tests.rs [hunk 4]
```diff
@@ -706,3 +708,118 @@ fn treasury_pallet_account_not_none() {
...
+		},
+		// DescribeAccountId32Terminal
+		TestCase {
+			description: "DescribeAccountId32Terminal Parent",
+			location: Location::new(
+				1,
+				[AccountId32 {
+					network: None,
+					id: polkadot_core_primitives::AccountId::from(ALICE).into(),
+				}],
```
[the remark this answers is not in the document]
Comment (x3c41a) [reply system-parachains/asset-hubs/asset-hub-kusama/tests/tests.rs]: ✔️

File: system-parachains/asset-hubs/asset-hub-kusama/tests/tests.rs (hunk 1 above)
Comment (x3c41a) [reply system-parachains/asset-hubs/asset-hub-kusama/tests/tests.rs]: ☑️
Out of curiosity, what edge-case you want to avoid by eliminating zero keys?

**Comment by x3c41a** [comment]
bot fmt

**Comment by x3c41a** [comment]
> The conversions are the same for all system parachains, so this test could be a macro to dedup the code, but I don't know where we would put that macro or if it's worth the effort. I'm fine with the duplicated code too.
> 
> Please see the other comments.

I don't feel like writing macros just yet, still want to get used to Rust.

**Review by acatangiu — APPROVED** [review]

**Review by franciscoaguirre — APPROVED** [review]

**Review by acatangiu — APPROVED** [review]

**Review by franciscoaguirre — APPROVED** [review]

**Comment by x3c41a** [comment]
Check Migrations  failures are unrelated

File: CHANGELOG.md [hunk 5]
```diff
@@ -12,6 +12,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - Update foreign asset ids in Asset Hub Polkadot and Asset Hub Kusama from v3 to v4 locations ([polkadot-fellows/runtimes#472](https://github.com/polkadot-fellows/runtimes/pull/472))
 - Lower Parachain and Data Deposits to Encourage Experimentation on Kusama ([polkadot-fellows/runtimes#501](https://github.com/polkadot-fellows/runtimes/pull/501))
 
+### Added
+
+- Location conversion tests for relays and parachains ([polkadot-fellows/runtimes#487](https://github.com/polkadot-fellows/runtimes/pull/487))
+
```
Comment (bkontur) [inline CHANGELOG.md:18]: @x3c41a 
move this at the begining under `## [Unreleased]` to unblock CI for migrations

Comment (x3c41a) [reply CHANGELOG.md]: It indeed helped to fix Migrations error. 
Thank you, Brani!

**Review by franciscoaguirre — APPROVED** [review]

File: system-parachains/people/people-polkadot/src/tests.rs [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+	let bob_20 = AccountKey20 { network: None, key: [123u8; 20] };
+
+	// the purpose of hardcoded values is to catch an unintended location conversion logic change.
+	struct TestCase {
+		description: &'static str,
+		location: Location,
+		expected_account_id_str: &'static str,
+	}
+
+	let test_cases = vec![
```
Comment (ggwpez) [inline system-parachains/people/people-polkadot/src/tests.rs:37]: Why are the test-cases not defined in a shared crate? They seem to be copy&pasted?

Comment (x3c41a) [reply system-parachains/people/people-polkadot/src/tests.rs:37]: There is a subtle difference among these tests -- DescribeAccountId32Terminal is different in many of the cases. 

If I were to refactor the tests I'd rather rewrite those into macros, as Adrian suggested a few comments before. I'll leave it as a follow-up for myself once I fully onboard.

Comment (x3c41a) [reply system-parachains/people/people-polkadot/src/tests.rs:37]: Unless you're talking about people folder specifically, though I would not do it for a single folder...

**Review by acatangiu — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]
